src/org/opensolaris/opengrok/index/Indexer.java

```diff
@@ -101,542 +109,658 @@ private static void A_usage() {
+            });
+
+            parser.on("-t", "", "=number", Integer.class,
+                "Default tab size to use (number of spaces per tab character).").Do( tabSize -> {
+                cfg.setTabSize((Integer)tabSize);
+            });
+
+            parser.on("-U", "--host", "=host:port", WebAddress.class,
+                "Send the current configuration to the specified address",
+                "(This is most likely the web-app configured with ConfigAddress)").
+                Do( webAddr -> {
+                    WebAddress web = (WebAddress)webAddr;
+
+                    env = RuntimeEnvironment.getInstance();
+
+                    host = web.getHost();
+                    port = web.getPort();
+                    env.setConfigHost(host);
+                    env.setConfigPort(port);
+                }
+            );
+
+            parser.on("---unitTest");  // For unit test only, will not appear in help
+
+            parser.on("-V", "--version", "Print version and quit.").Do( v -> {
+                System.out.println(Info.getFullVersion());
+                System.exit(0);
+            });
+
+            parser.on("-v", "--verbose", "Print progress information as we go along.").Do( v -> {
+                cfg.setVerbose(true);
+                LoggerUtil.setBaseConsoleLogLevel(Level.INFO);
+            });
+
+            parser.on("-W", "--writeConfig", "=/path/to/configuration", 
+                "Write the current configuration to the specified file",
+                "(so that the web application can use the same configuration").Do( configFile -> {
+                configFilename = (String)configFile;
+            });
+
+            parser.on("-w", "--web", "=webapp-context", 
+                "Context of webapp. Default is /source. If you specify a different name,",
+                "make sure to rename source.war to that name. Also FULL reindex is needed",
+                "if this is changed.").
+                Do( webContext -> {
+                    String webapp = (String)webContext;
+                    if (webapp.charAt(0) != '/' && !webapp.startsWith("http")) {
+                        webapp = "/" + webapp;
+                    }
+                    if (webapp.endsWith("/")) {
+                        cfg.setUrlPrefix(webapp + "s?");
                     } else {
-                        getInstance().sendToConfigHost(env, host, port);
+                        cfg.setUrlPrefix(webapp + "/s?");
                     }
                 }
-            } catch (IndexerException ex) {
-                LOGGER.log(Level.SEVERE, "Exception running indexer", ex);
-                System.err.println(cmdOptions.getUsage());
-                System.exit(1);
-            } catch (Throwable e) {
-                System.err.println("Exception: " + e.getLocalizedMessage());
-                LOGGER.log(Level.SEVERE, "Unexpected Exception", e);
+            );
+
+            parser.on("-X", "=url:suffix", 
```

Comment:
ditto

Reply:
Do you have a suggestion for the long version name for this one? I couldn't think of one at the time.

Reply:
`--user-page-suffix`

Reply:
OK, I noticed that you are using hyphens to separate the names as opposed to CamelCase.  Do you want me to go back and do the same for the other options too?

Reply:
That's a good question. I am not strongly bound for either, feel free to stick with CamelCase.